sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue

```diff
@@ -281,13 +290,30 @@ export default defineComponent({
       applicationWizard = form;
     };
 
-    const pageChanged = (
+    const pageChanged = async (
       isInFirstPage: boolean,
       currentPage: number,
       isInLastPage: boolean,
     ) => {
       isFirstPage.value = isInFirstPage;
       isLastPage.value = isInLastPage;
+      if (!savedDraftData) {
+        savedDraftData = JSON.stringify(applicationWizard.submission.data);
+      }
+      const dataChanged = savedDraftData !== applicationWizard.submission.data;
+      if (
+        !notDraft.value &&
+        !isReadOnly.value &&
+        !isFirstPage.value &&
+        !submittingApplication.value &&
+        !automaticDraftSaveInProgress.value &&
+        !savingDraft.value &&
+        dataChanged
+      ) {
+        automaticDraftSaveInProgress.value = true;
+        await saveDraft();
+        automaticDraftSaveInProgress.value = false;
+      }
```

Comment:
As suggested during the sync up. I would recommend this code snippet which will make the logic organized. 

```ts
const callSaveDraft = async () => {      const associatedFiles = formioUtils.getAssociatedFiles(applicationWizard);      await ApplicationService.shared.saveApplicationDraft(props.id, {        programYearId: props.programYearId,        data: applicationWizard.submission.data,        associatedFiles,      });    };    // Save the current state of the student application skipping all validations.    const saveDraft = async () => {      savingDraft.value = true;      try {        await callSaveDraft();        snackBar.success("Application draft saved with success.");      } catch {        snackBar.error("An unexpected error has happened.");      } finally {        savingDraft.value = false;      }    };    const saveDraftAutomatically = async () => {      automaticDraftSaveInProgress.value = true;      try {        await callSaveDraft();        savedDraftData = JSON.stringify(applicationWizard.submission.data);      } finally {        automaticDraftSaveInProgress.value = false;      }    };
```

Reply:
Modified and updated it as per our discussion.

![image](https://github.com/user-attachments/assets/3d6cf1a5-3376-46be-a790-fd01fa6b5132)